src/context/store.ts

```diff
@@ -1,15 +1,67 @@
-import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
+import { Map } from 'immutable';
+import {
+  configureStore,
+  getDefaultMiddleware,
+  combineReducers,
+} from '@reduxjs/toolkit';
+import {
+  innerReducer,
+  outerReducer,
+  middleware,
+} from 'redux-async-initial-state';
+
 import mapStateReduce from './mapStateSlice';
+import serverStateReduce from './serverStateSlice';
+import { getLayersAvailableDates } from '../utils/server-utils';
+import { LAYERS_AVAILABLE_DATES } from '../constants';
+import { AvailableDates } from '../config/types';
 
-export const store = configureStore({
-  reducer: {
+const initializeStore = async (getCurrentState: any) => {
+  const isStorageSet = localStorage.getItem(LAYERS_AVAILABLE_DATES) !== null;
+
+  if (isStorageSet) {
+    // Aborting the initialization
+    return Promise.reject();
+  }
+
+  const layersAvailableDates = await getLayersAvailableDates();
+  const currentState = getCurrentState();
+  const { serverState } = currentState;
+
+  localStorage.setItem(
+    LAYERS_AVAILABLE_DATES,
+    JSON.stringify(layersAvailableDates),
+  );
+
+  return new Promise(resolve => {
+    resolve({

```

Comment:
This doesn't actually do anything - the return from an `async` function is already wrapped in a Promise, just return the object.

Reply:
Without the returned `Promise`, I am having an error.
Looks like it's needed (cf [docs](https://github.com/KELiON/redux-async-initial-state#partial-replace))